moto/rds2/models.py

```diff
@@ -983,29 +1158,46 @@ def create_database(self, db_kwargs):
         self.databases[database_id] = database
         return database
 
-    def create_snapshot(
+    def create_database_snapshot(
         self, db_instance_identifier, db_snapshot_identifier, tags=None
     ):
         database = self.databases.get(db_instance_identifier)
         if not database:
             raise DBInstanceNotFoundError(db_instance_identifier)
-        if db_snapshot_identifier in self.snapshots:
+        if db_snapshot_identifier in self.database_snapshots:
             raise DBSnapshotAlreadyExistsError(db_snapshot_identifier)
-        if len(self.snapshots) >= int(os.environ.get("MOTO_RDS_SNAPSHOT_LIMIT", "100")):
+        if len(self.database_snapshots) >= int(os.environ.get("MOTO_RDS_SNAPSHOT_LIMIT", "100")):
             raise SnapshotQuotaExceededError()
         if tags is None:
             tags = list()
         if database.copy_tags_to_snapshot and not tags:
             tags = database.get_tags()
-        snapshot = Snapshot(database, db_snapshot_identifier, tags)
-        self.snapshots[db_snapshot_identifier] = snapshot
+        snapshot = DatabaseSnapshot(database, db_snapshot_identifier, tags)
+        self.database_snapshots[db_snapshot_identifier] = snapshot
         return snapshot
 
-    def delete_snapshot(self, db_snapshot_identifier):
-        if db_snapshot_identifier not in self.snapshots:
+    def copy_database_snapshot(
+        self, source_snapshot_identifier, target_snapshot_identifier, tags=None,
+    ):
+        if source_snapshot_identifier not in self.database_snapshots:
+            raise DBSnapshotNotFoundError(source_snapshot_identifier)
+        if target_snapshot_identifier in self.database_snapshots:
+            raise DBSnapshotAlreadyExistsError(target_snapshot_identifier)
+        if len(self.database_snapshots) >= int(os.environ.get("MOTO_RDS_SNAPSHOT_LIMIT", "100")):
+            raise SnapshotQuotaExceededError()
+        source_snapshot = self.database_snapshots[source_snapshot_identifier]
+        target_snapshot = copy.deepcopy(source_snapshot)
+        target_snapshot.snapshot_id = target_snapshot_identifier
+        if tags is not None:
+            target_snapshot.tags += tags
```

Comment:
What happens if the user provides a tag that already exists? Does it persist and return duplicate tags?
That's probably not what AWS does, so it would be good to have a test case for this as well, to ensure it behaves as it should